encoding/hex.ts

```diff
@@ -85,18 +105,20 @@ export function decode(dest: Uint8Array, src: Uint8Array): [number, Error] {
   return [i, undefined];
 }
 
-// DecodedLen returns the length of a decoding of x source bytes.
-// Specifically, it returns x / 2.
+/**
+ * DecodedLen returns the length of a decoding of `x` source bytes. Specifically, it returns `x / 2`.
+ * @param x
+ */
 export function decodedLen(x: number): number {
-  return x / 2;
+  return Math.floor(x / 2);
 }
 
-// DecodeString returns the bytes represented by the hexadecimal string s.
-//
-// DecodeString expects that src contains only hexadecimal
-// characters and that src has even length.
-// If the input is malformed, DecodeString returns
-// the bytes decoded before the error.
+/**
+ * DecodeString returns the bytes represented by the hexadecimal string `s`.
+ * DecodeString expects that src contains only hexadecimal characters and that src has even length.
+ * If the input is malformed, DecodeString will throws an error.
+ * @param s
```

Comment:
You only need to add this if you are actually going to add a comment about the argument, otherwise I would just omit it as it adds nothing.